src/placeos-models/edge.cr

```diff
@@ -0,0 +1,27 @@
+require "random"
+
+require "./base/model"
+
+module PlaceOS::Model
+  class Edge < ModelBase
+    include RethinkORM::Timestamps
+
+    table :edge
+
+    attribute name : String, es_subfield: "keyword"
+    attribute secret : String = ->{ Random::Secure.hex(64) }
```

Comment:
This may be a discussion better suited to elsewhere, but is a straight up name + fixed secret the best option for access control? They will likely be running on infrastructure that sits outside of any isolated inter-service networks so are a little more exposed. Assuming that they have access to decrypted settings, this could be a weak point. An alternative is to treat them the same as any external client and require short lived tokens and entry via the main API service.

If fixed is the best (even if it's just temp for an initial impl), probably shouldn't keep it in clear text.

Reply:
I'll drop in encryption of the secret